### ISSUE TITLE
Add `Relational.exists` function, and push it down to Calcite

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#51]
Add `Relational.exists` function, which can be used as `exists` in queries, like this:
```
from d in scott.dept
where exists (
  from e in scott.emp
  where e.deptno = d.deptno)
```

Push `exists` down to Calcite relational algebra (typically as a semi-join). Queries typically have correlating variables (the above query has `d.deptno`, for an example) so we will need to deal with those also.

Related are the `elem` and `notElem` infix operators, which are analogous to SQL's `IN` and `NOT IN` (but without SQL's horrible null-semantics). They need to be pushed down to Calcite, too, but we probably won't deal with those in this case.

**Comment by julianhyde** [comment]
`exists list` is a straightforward function: it returns whether the list is non-empty. Adding the built-in function is trivial, but the push-down to Calcite is the hard part.

Note the `List.exists` and `Vector.exists` functions are rather different: they check whether an element in a list or vector matches a given predicate. It is unfortunate that the names are similar.

Since `Relational.exists` operates on a list, it can be used as an aggregate function. It's not very useful, because the set of rows going into an aggregate function in `group` is always non-empty, and therefore `exists` will always return true. If we implement the equivalent of SQL's `FILTER` (filtered aggregate functions) or `OVER` (windowed aggregate functions), `exists` may become more useful.

**Comment by julianhyde** [comment]
Fixed in d0f9175.
